tests/algorithms/cca_tests.cpp

```diff
@@ -0,0 +1,170 @@
+                0.01, 0.0001 * std::max(io_truth.channel0, io_truth.channel1));
+
+            auto match = std::find_if(
+                meas.begin(), meas.end(),
+                [&io_truth, tol](const traccc::measurement &i) {
+                    return std::abs(i.local[0] - io_truth.channel0) < tol &&
+                           std::abs(i.local[1] - io_truth.channel1) < tol;
+                });
+
+            ASSERT_TRUE(match != meas.end());
+
+            EXPECT_NEAR(match->local[0], io_truth.channel0, tol);
+            EXPECT_NEAR(match->local[1], io_truth.channel1, tol);
+            EXPECT_NEAR(match->variance[0], io_truth.variance0, tol);
+            EXPECT_NEAR(match->variance[1], io_truth.variance1, tol);
+
+            ++total_truth;
+        }
+
+        EXPECT_EQ(total_truth, total_found);
+    }
+};
+
+TEST_P(ConnectedComponentAnalysisTests, CpuSparseCCLAlgorithm) {
+    traccc::component_connection cc;
+    traccc::measurement_creation mc;
+    traccc::cell_module module{.module = 0, .pixel = {.0, .0, .0, .0}};
+
+    std::function<traccc::host_measurement_collection(
+        const traccc::host_cell_collection &)>
+        fp = traccc::compose(std::function<traccc::cluster_collection(
+                                 const traccc::host_cell_collection &)>(
+                                 std::bind(cc, std::placeholders::_1, module)),
+                             std::function<traccc::host_measurement_collection(
+                                 const traccc::cluster_collection &)>(
+                                 std::bind(mc, std::placeholders::_1, module)));
+
+    std::function<
+        std::map<traccc::geometry_id, std::vector<traccc::measurement>>(
+            const traccc::host_cell_container &)>
+        f = [fp](const traccc::host_cell_container &data) {
+            std::map<traccc::geometry_id, std::vector<traccc::measurement>>
+                result;
+
+            for (std::size_t i = 0; i < data.size(); ++i) {
+                traccc::host_measurement_collection measurements =
+                    fp(data.at(i).items);
+                std::vector<traccc::measurement> out(measurements.begin(),
+                                                     measurements.end());
+                result.emplace(data.at(i).header.module, std::move(out));
+            }
+
+            return result;
+        };
+
+    test_connected_component_analysis(GetParam(), f);
+}
+
+INSTANTIATE_TEST_SUITE_P(
+    , ConnectedComponentAnalysisTests, ::testing::ValuesIn(get_test_files()),
+    [](const testing::TestParamInfo<ConnectedComponentAnalysisTests::ParamType>
+           &info) { return info.param; });
+
+// Google Test can be run manually from the main() function
+// or, it can be linked to the gtest_main library for an already
+// set-up main() function primed to accept Google Test test cases.
+int main(int argc, char **argv) {
+    ::testing::InitGoogleTest(&argc, argv);
+
+    return RUN_ALL_TESTS();
+}
```

Comment:
As I keep pestering @beomki-yeo as well, we should not be implementing `main(...)` functions for our tests. If we don't, if we just use `GTest::gtest_main`, then we can freely choose in the CMake configuration how we want to link our tests into executables.

In this setup this test must always be compiled into a standalone executable, it can't be combined with other tests into a larger executable. (Which is a design that I really like about VecMem's tests...)

Reply:
That's fair, I think we have a lot of standardization to do for traccc tests in general, something we can look into as a chore.